src/services/user.service.ts

```diff
@@ -0,0 +1,125 @@
+import { ChatAPI, UserAPI } from "api";
+import type { Dispatch } from "core";
+import { transformUser, apiHasError } from "utils";
+import { RegisterDTO as UserDataPayload, UserDTO } from "types/api";
+import { fileToFormData } from "utils";
+
+export const updateUserData = async (
+  dispatch: Dispatch<AppState>,
+  state: AppState,
+  payload: UserDataPayload
+) => {
+  dispatch({ isLoading: true, formSuccess: null });
+  const api = new UserAPI();
+  const response = await api.updateUserData(payload);
+  if (apiHasError(response)) {
+    dispatch({ isLoading: false, formError: response.reason });
+    return;
+  }
+  dispatch({
+    isLoading: false,
+    formError: null,
+    formSuccess: "Данные успешно обновлены",
+    user: transformUser(response)
+  });
+};
+
+export const updateUserPassword = async (
+  dispatch: Dispatch<AppState>,
+  state: AppState,
+  payload: UserDataPayload
+) => {
+  dispatch({ isLoading: true, formSuccess: null });
+  const api = new UserAPI();
+  const response = await api.updateUserPassword(payload);
+  if (apiHasError(response)) {
+    dispatch({ isLoading: false, formError: response.reason });
+    return;
+  }
+  dispatch({
+    isLoading: false,
+    formError: null,
+    formSuccess: "Пароль успешно обновлен"
+  });
+};
+
+export const updateUserAvatar = async (
+  dispatch: Dispatch<AppState>,
+  state: AppState,
+  payload
```

Comment:
Тут можно типизировать

```suggestion
  payload: File
```